src/fileio.cpp

```diff
@@ -429,6 +429,10 @@ FILE *FioFOpenFile(const std::string &filename, const char *mode, Subdirectory s
 			if (token == "..") {
 				if (tokens.size() < 2) return nullptr;
 				tokens.pop_back();
+			} else if (token == ".") {
+				/* Do nothing.
+				 * Useful if a tar contains a folder named '.', which then causes weird file resolving issues,
+				 * like changelog.txt from the current working directory, rather than within the tar. */
```

Comment:
This comment is a bit weird :P "Useful" seems to be misused here. And it continues on to a specific situation, which no future-us will understand. What has "changelog.txt" have to do with anything in this place in the code. That only makes sense if you come from the bug-report.

`/* Do nothing. "." means current folder, but you can create tar files with "." in the path. This confuses our file resolver. So, act like this folder doesn't exist. */`

Or something?